datapackage_pipelines_ckan/processors/dump/to_ckan.py

```diff
@@ -0,0 +1,218 @@
+import os
+import json
+import hashlib
+
+import datapackage as datapackage_lib
+from ckan_datapackage_tools import converter
+from datapackage_pipelines.lib.dump.dumper_base import FileDumper, DumperBase
+
+from datapackage_pipelines_ckan.utils import make_ckan_request
+
+import logging
+log = logging.getLogger(__name__)
+
+
+class CkanDumper(FileDumper):
+
+    def initialize(self, parameters):
+        super(CkanDumper, self).initialize(parameters)
+        self.ckan_base_url = '{ckan_host}/api/3/action/'.format(
+            ckan_host=parameters['ckan-host'])
+        self.ckan_api_key = parameters.get('ckan-api-key')
+        self.dataset_resources = []
+        self.dataset_id = None
+
+    def handle_resources(self, datapackage,
+                         resource_iterator,
+                         parameters, stats):
+        '''
+            This method handles most of the control flow for creating
+            dataset/resources on ckan.
+
+            First we handle the dataset creation, then, if successful, we
+            create each resource.
+        '''
+
+        # Calculate datapackage hash
+        if self.datapackage_hash:
+            datapackage_hash = hashlib.md5(
+                        json.dumps(datapackage,
+                                   sort_keys=True,
+                                   ensure_ascii=True).encode('ascii')
+                    ).hexdigest()
+            DumperBase.set_attr(datapackage, self.datapackage_hash,
+                                datapackage_hash)
+
+        # Handle the datapackage first!
+        self.handle_datapackage(datapackage, parameters, stats)
+
+        # Handle non-streaming resources
+        for resource in datapackage['resources']:
+            if not resource.get('dpp:streaming', False):
+                resource_metadata = {
+                    'package_id': self.dataset_id,
+                    'url': resource['dpp:streamedFrom'],
+                    'name': resource['name'],
```

Comment:
Did you leave out `format` on purpose? CKAN will try to guess it from the file name, but I'm not sure if this applies to DPP outputs

Reply:
I'll add it, if it's in the datapackage resource dict.